ui/src/app/job-details/resources/resources.component.ts

```diff
@@ -43,28 +43,23 @@ export class JobResourcesComponent implements OnInit {
     private readonly viewContainer: ViewContainerRef) {}
 
   ngOnInit() {
-    if (this.job.inputs) {
+    if (this.job.inputs && Object.keys(this.job.inputs).length > 0) {
       this.inputs = Object.keys(this.job.inputs).sort();
       this.tabIds.push('inputs');
     }
-    if (this.job.outputs) {
+    if (this.job.outputs && Object.keys(this.job.outputs).length > 0) {
       this.outputs = Object.keys(this.job.outputs).sort();
       this.tabIds.push('outputs');
     }
 
     if (this.job.extensions) {
-      if (this.job.extensions.sourceFile) {
+      if (this.job.extensions.sourceFile && this.job.extensions.sourceFile.length > 0) {
         this.sourceFile = this.job.extensions.sourceFile;
         this.tabIds.push('source-file');
       }
 
       if (this.job.extensions.logs) {
-        for (let file of Object.keys(this.job.extensions.logs)) {
-          this.readResourceFile(file);
-          let tabId = 'log-' + file;
-          this.tabIds.push(tabId);
-          this.tabTitles.set(tabId, file);
-        }
+        Object.keys(this.job.extensions.logs).forEach(file => this.readResourceFile(file))
```

Comment:
I would rework this slightly. The issue is that the log files could load in in an arbitrary order. Using a a `Promise.all()` on the `readResourceFile` promises, then pushing tabIds in a for loop in a single callback would resolve this.

Reply:
Good point, switched to that approach.